graphprot/NeuralNet.py

```diff
@@ -191,42 +193,55 @@ def plot_hit_rate(self, data='eval', threshold=4, mode='percentage'):
             plt.xlabel("Number of models")
             plt.ylabel("Hit Rate")
             plt.legend()
-            plt.savefig('hitrate.png')
+            plt.savefig('hitrate{}.png'.format(name))
             plt.close()
 
         except:
             print('No hit rate plot could be generated for you {} task'.format(
                 self.task))
 
-    def train(self, nepoch=1, validate=False, plot=False):
+    def train(self, nepoch=1, validate=False, plot=False, save_model='last', hdf5='train_data.hdf5', save_epoch='intermediate'):
         """Train the model
 
         Args:
             nepoch (int, optional): number of epochs. Defaults to 1.
             validate (bool, optional): perform validation. Defaults to False.
             plot (bool, optional): plot the results. Defaults to False.
+            savemodel (last, best): save the model. Defaults to 'last' 
         """
 
+        # Output file 
+        fname = os.path.join(self.outdir, hdf5)
+        self.f5 = h5py.File(fname, 'w')
```

Comment:
by default this will erase the file fname if it exists which can be dangerous if you rerun an experiment but want to keep the previous results. I would add a check to see if the file exists and if it does change the name with a number. train_data_001.hdf5

Reply:
Done ! If train_data_001.hdf5 exists, then I change it to train_data_002.hdf5 and so on.

Reply:
great !